cmd/sealer/cmd/cluster/upgrade.go

```diff
@@ -0,0 +1,206 @@
+// Copyright © 2021 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cluster
+
+import (
+	"fmt"
+	"github.com/sealerio/sealer/common"
+	v12 "github.com/sealerio/sealer/pkg/define/image/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"path/filepath"
+
+	"github.com/sealerio/sealer/cmd/sealer/cmd/types"
+	"github.com/sealerio/sealer/cmd/sealer/cmd/utils"
+	clusterruntime "github.com/sealerio/sealer/pkg/cluster-runtime"
+	"github.com/sealerio/sealer/pkg/clusterfile"
+	imagecommon "github.com/sealerio/sealer/pkg/define/options"
+	"github.com/sealerio/sealer/pkg/imagedistributor"
+	"github.com/sealerio/sealer/pkg/imageengine"
+	"github.com/sealerio/sealer/pkg/infradriver"
+	"sigs.k8s.io/yaml"
+)
+
+var upgradeFlags *types.UpgradeFlags
+
+var longUpgradeCmdDescription = `upgrade command is used to upgrade cluster.
```

Comment:
if this cluster is not installed by sealer , can we upgrade it ?

Reply:
not supported currently